### PR TITLE
Fix restart_period description in CONTRIBUTING.md

### LINKED ISSUE #46: Error in contribution guide
The contribution guide suggests restart periods but uses the pandas frequency notation that is used for `restart_freq`. 

It should be updated to use values in the format that are used in the namelist.

### PR DESCRIPTION
**Pull request by aidanheerdegen**
Fixes #46

### CODE REVIEW

**Comment by aidanheerdegen** [comment]
Just need someone to run an eye over this to check if there is anything else wrong that I've missed.

**Comment by aidanheerdegen** [comment]
We should also add some information about the ice concatenation script not working unless `sync` is enabled. 

See also https://github.com/ACCESS-NRI/access-om2-configs/issues/45#issuecomment-2034219731

**Comment by aidanheerdegen** [comment]
Something we've overlooked: propagating automated commits in `release-*` branches, when versions (and checksums) are updated, back to `dev-*` branches. This is necessary so that subsequent PRs from `dev-*` are up to to date with `release-*`.

So we need a step to `git  pull origin release-*` into the `dev-*` branch.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -159,4 +161,4 @@ The requirement is simply that a date-based frequency be used so that restarts a
 
 **sync**
 
-This should not be enabled. Nor should `sync_path` be set to a real path. Ideally set `sync_path` to `null`.
\ No newline at end of file
+This should not be enabled. Nor should `sync_path` be set to a real path. Ideally set `sync_path` to `null`.
```
Comment (jo-basevi) [inline CONTRIBUTING.md:164]: Maybe use the following as an example as it's `path` rather than `sync_path`
```
sync:
    enable: false
    path: null
```

Comment (aidanheerdegen) [reply CONTRIBUTING.md:164]: Thanks.

**Comment by aidanheerdegen** [comment]
> Something we've overlooked: propagating automated commits in `release-*` branches, when versions (and checksums) are updated, back to `dev-*` branches. This is necessary so that subsequent PRs from `dev-*` are up to to date with `release-*`.
> 
> So we need a step to `git pull origin release-*` into the `dev-*` branch.

Actually now that I think about it, this isn't a problem when making a PR from `dev-*` to `release-*` as the version bump and checksum updates are done on the PR branch, which is `dev-*`. 

It is an issue when the initial checksum workflow is used, as that makes changes only on `release-*`.

**Comment by CodeGat** [comment]
> Actually now that I think about it, this isn't a problem when making a PR from `dev-*` to `release-*` as the version bump and checksum updates are done on the PR branch, which is `dev-*`.
> 
> It is an issue when the initial checksum workflow is used, as that makes changes only on `release-*`.

Correct.

**Review by CodeGat — APPROVED** [review]

**Pull request merged**